src/pages/UserPage/UserPage.tsx

```diff
@@ -29,6 +35,23 @@ const UserPage: FC = () => {
 
   const userNameString = `${user.user.forename} ${user.user.surname}`
 
+  const handleArchiveModal = () => {
+    const isMainUser = some(user?.roles, (mainUser) => mainUser?.is_root)
```

Comment:
I you have only on key then you can use shorthand version: 'some(user?.roles, 'is_root')'